ext/snmp/snmp.c

```diff
@@ -674,25 +673,33 @@ static int php_snmp_parse_oid(
 		objid_query->vars = (snmpobjarg *)emalloc(sizeof(snmpobjarg));
 		objid_query->vars[objid_query->count].oid = ZSTR_VAL(oid_str);
 		if (st & SNMP_CMD_SET) {
-			if (type_str && value_str) {
-				if (ZSTR_LEN(type_str) != 1) {
-					php_error_docref(NULL, E_WARNING, "Bogus type '%s', should be single char, got %zu", ZSTR_VAL(type_str), ZSTR_LEN(type_str));
-					efree(objid_query->vars);
-					return FALSE;
-				}
-				pptr = ZSTR_VAL(type_str);
-				objid_query->vars[objid_query->count].type = *pptr;
-				objid_query->vars[objid_query->count].value = ZSTR_VAL(value_str);
-			} else {
-				php_error_docref(NULL, E_WARNING, "Single objid and multiple type or values are not supported");
+			if (type_ht) {
+				zend_type_error("Type must be of type string when object ID is a string");
+				efree(objid_query->vars);
+				return FALSE;
+			}
+			if (value_ht) {
+				zend_type_error("Value must be of type string when object ID is a string");
 				efree(objid_query->vars);
 				return FALSE;
 			}
+
+			/* Both type and value must be valid strings */
+			ZEND_ASSERT(type_str && value_str);
+
+			if (ZSTR_LEN(type_str) != 1) {
+				zend_value_error("Type must be a single character");
+				efree(objid_query->vars);
+				return FALSE;
+			}
+			pptr = ZSTR_VAL(type_str);
+			objid_query->vars[objid_query->count].type = *pptr;
+			objid_query->vars[objid_query->count].value = ZSTR_VAL(value_str);
 		}
 		objid_query->count++;
 	} else if (oid_ht) { /* we got objid array */
 		if (zend_hash_num_elements(oid_ht) == 0) {
-			php_error_docref(NULL, E_WARNING, "Got empty OID array");
+			zend_value_error("Object ID cannot be an empty array");
```

Comment:
```suggestion
			zend_value_error("Object ID array cannot be empty");
```
or
```suggestion
			zend_value_error("Array of object IDs cannot be empty");
```

An object ID itself isn't an array.